### DISCUSSION TITLE
os.iorunv获取的返回值作为头文件路径为什么包含了换行符?

### DISCUSSION THREAD

**Discussion by rookiooO** [xmake-io/xmake#4594] [Q&A] [answered]
萌新一枚，第一次使用xmake，lua也不太熟悉。
具体代码如下：
local path,err = os.iorunv("/usr/bin/pg_config", {"--includedir-server"})
target:add({includedirs  = path})
编译时会报错说找不到头文件，查看详细编译信息时，-I中已经包含了我指定路径，仔细一看打印信息中-I/user/include/postgresql/16/server后面多了一个换行。用string.gsub删除换行符后，编译成功。
想请问一下，有没有更好的办法获取pg_config提供的路径信息，而不需要手动去删除换行符。

**Comment by waruqi** [comment, marked as answer]
xmake 本身不会追加换行，那是 pg_config 输出就带了的，不是 xmake 的问题

你要去掉，就自己 trim 下。`result = result:trim()`

另外，path 是内置模块，不要用 path 命名的变量

**Reply by rookiooO** [reply]
大佬，谢谢啦，刚刚仔细看了下原来cmake的execute_process里面填了一个去除结尾换行的参数，是我疏忽了。
您是作者么，我觉得xmake真的不错。希望能越来越好！！！

**Reply by rookiooO** [reply]
以前都是用cmake都是用别人写好的find脚本就没有注意了
